samples/batch/csharp/program.cs

```diff
@@ -6,136 +6,155 @@
-                // for each transcription in the list we check the status
-                foreach (var transcription in transcriptions)
+
+                // get all transcriptions for the user
+                paginatedTranscriptions = null;
+                do
                 {
-                    switch (transcription.Status)
+                    // <transcriptionstatus>
+                    if (paginatedTranscriptions == null)
+                    {
+                        paginatedTranscriptions = await client.GetTranscriptionsAsync().ConfigureAwait(false);
+                    }
+                    else
                     {
-                        case "Failed":
-                        case "Succeeded":
-                            // we check to see if it was one of the transcriptions we created from this client.
-                            if (!createdTranscriptions.Contains(transcription.Id))
-                            {
-                                // not created form here, continue
-                                continue;
-                            }
-                            completed++;
-
-                            // if the transcription was successful, check the results
-                            if (transcription.Status == "Succeeded")
-                            {
-                                var resultsUri0 = transcription.ResultsUrls["channel_0"];
-
-                                WebClient webClient = new WebClient();
-
-                                var filename = Path.GetTempFileName();
-                                webClient.DownloadFile(resultsUri0, filename);
-                                var results0 = File.ReadAllText(filename);
-                                var resultObject0 = JsonConvert.DeserializeObject<RootObject>(results0);
-                                Console.WriteLine(results0);
-
-                                Console.WriteLine("Transcription succeeded. Results: ");
-                                Console.WriteLine(results0);
-                            }
-                            else
-                            {
-                                Console.WriteLine("Transcription failed. Status: {0}", transcription.StatusMessage);
-                            }
-                            break;
-
-                        case "Running":
-                            running++;
-                            break;
-
-                        case "NotStarted":
-                            notStarted++;
-                            break;
+                        paginatedTranscriptions = await client.GetTranscriptionsAsync(paginatedTranscriptions.NextLink).ConfigureAwait(false);
                     }
+
+                    // delete all pre-existing completed transcriptions. If transcriptions are still running or not started, they will not be deleted
+                    foreach (var transcription in paginatedTranscriptions.Values)
```

Comment:
Is this a pattern we actually want users to follow? Don't we rather want them to create subscription, and then query its status directly, instead of polling all the transcriptions constantly? In the sample we delete all existing transcriptions before, but altogether it doesn't seem to be the most elegant thing to show first-time users. (My 0.02$)

Reply:
Yes, this is planned for the next set of changes to this doc. The first goal was to update it to v3 with same set of of functionality exercised.